core/src/main/java/com/scalar/db/common/error/SubCategory.java

```diff
@@ -0,0 +1,45 @@
+package com.scalar.db.common.error;
+
+import java.util.Objects;
+
+public enum SubCategory {
+
+  //
+  // Sub-categories for retryable errors
+  //
+  TRANSACTION_CONFLICT(Category.RETRYABLE_ERROR, "0"),
+  TRANSACTION_NOT_FOUND(Category.RETRYABLE_ERROR, "1"),
+
+  //
+  // Sub-categories for user errors
+  //
+  ILLEGAL_ARGUMENT(Category.USER_ERROR, "0"),
+  AUTHENTICATION_ERROR(Category.USER_ERROR, "1"),
+  AUTHORIZATION_ERROR(Category.USER_ERROR, "2"),
+
+  //
+  // Sub-categories for internal errors
+  //
+  UNKNOWN_TRANSACTION_STATUS(Category.INTERNAL_ERROR, "0"),
```

Comment:
Probably users want to know whether to retry the failed transaction operations based on the category in error code. There is the separate `RETRYABLE_ERROR` category, so I guess users shouldn't retry for `INTERNAL_ERROR`? I know `UNKNOWN_TRANSACTION_STATUS` shouldn't be retried to avoid duplicated inserts or etc., but I'm wondering what kind of errors would be categorized in `UNKNOWN`.

(This isn't a suggestion from me and I'm just writing down my random thoughts)

Maybe connection failures to the database is categorized in `INTERNAL_ERROR` via `UNKNOWN`. In this case, I think retry would work with some interval and `RETRYABLE_ERROR` may be an option.

But if the connection error occurred since the database was busy, the retry interval should be longer than the retry for transaction conflicts. Considering this case, 2 retryable categories are needed for transaction conflicts and other tentative errors...? 🤔

Reply:
Firstly, the purpose of this PR is just to add error codes to error messages, so I think we can address error handling as a separate concern. However, it's true that these aspects can't be entirely isolated actually.

I think that when we encounter `RETRYABLE_ERROR`, we can retry the transaction for sure. However, when we face `INTERNAL_ERROR` via `UNKNOWN`, it's uncertain whether we can retry the transaction. In some cases, we may be able to retry the transaction, but in others, we may not. From this perspective, categorizing such errors as `UNKNOWN` seems reasonable.

Reply:
As discussed, I will rename `RETRYABLE_ERROR` to `TRANSACTION_RETRYABLE_ERROR` to imply that users could retry even for `INTERNAL_ERROR`.

Reply:
Fixed in 60fb835a531e657dc7f1e8485aa2a03650819056. Thanks.